internal/components/componenttest/assert_no_error_host.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package componenttest is a temporary addition while related changes are not made on core.
+package componenttest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// assertNoErrorHost implements a component.Host that asserts that
+// there were no errors.
+type assertNoErrorHost struct {
```

Comment:
Could be helpful to have this exported so it can be embedded elsewhere now that the main nophost was unexported.

Reply:
I will postpone this for the time being.